thoth/storages/graph/dgraph.py

```diff
@@ -2029,6 +2056,73 @@ def sync_analysis_result(self, document: dict) -> None:
         self._python_sync_analysis_result(package_extract_run, document, environment)
         self._python_file_digests_sync_analysis_result(package_extract_run, document)
 
+    @enable_vertex_cache
+    def sync_package_analysis_result(self, document: dict) -> None:
+        """Sync the given package analysis result to the graph database."""
+        package_analysis_document_id = PackageAnalysisResultsStore.get_document_id(document)
+        package_name = document["metadata"]["arguments"]["python"]["package_name"]
+        package_version = document["metadata"]["arguments"]["python"]["package_version"]
+        index_url = document["metadata"]["arguments"]["python"]["index_url"]
+
+        # TODO: capture errors on package analysis? result of package-analyzer should be a JSON with error flag
+        package_analyzer_run = PackageAnalyzerRun.from_properties(
+            package_analysis_document_id=package_analysis_document_id,
+            package_analysis_datetime=document["metadata"]["datetime"],
+            package_analyzer_version=document["metadata"]["analyzer_version"],
+            package_analyzer_name=document["metadata"]["analyzer"],
+            debug=document["metadata"]["arguments"]["thoth-package-analyzer"]["verbose"],
+            package_analyzer_error=False,
+            duration=None,  # TODO: assign duration
+        ).get_or_create(self.client)
+
+        package_index = PythonPackageIndex.query_one(self.client, url=index_url)
+        if not package_index:
+            raise PythonIndexNotRegistered(
+                f"Cannot insert PythonPackageVersionEntity record into database, "
+                f"no Python index with url {index_url} registered"
+            )
+
+        entity, _ = self.create_python_package_version_entity(package_name, package_version, index_url)
+        ProvidedBy.from_properties(
+            source=entity,
+            target=python_index,
+        ).get_or_create(self.client)
+
+        PackageAnalyzerInput.from_properties(
+            source=entity,
+            target=package_analyzer_run,
+        ).get_or_create(self.client)
+
+        for artifact in document["result"][index_url]["artifacts"]:
+            python_artifact = PythonArtifact.from_properties(
+                artifact_hash_sha256=artifact["sha256"],
+                artifact_name=artifact["name"],
+            ).get_or_create(self.client)
+
+            InvestigatedBy.from_properties(
+                source=package_analyzer_run,
+                target=python_artifact,
+            ).get_or_create(self.client)
+
+            for digest in artifact["digests"]:
+                filepath = digest["filepath"]
+                if filepath.endswith(".py"):
```

Comment:
Please create a follow-up else branch where there will be logged a warning about not synced files which are not python files so we see them in logs.